### PR TITLE
Add the model post-processing function

### PR DESCRIPTION
**Pull request by zhiqiangdon**
Add `model_postprocess_fn` to process the model output for inference. This is to handle the cases where the training loss function includes some operations that are also required to transform predictions in inference. For example, if we use the `nn.BCEWithLogitsLoss` on a regression task, we need to use sigmoid to transform the logits in inference.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/models/utils.py [hunk 1]
```diff
@@ -543,3 +545,44 @@ def get_hf_config_and_model(checkpoint_name: str, pretrained: Optional[bool] = T
         model = AutoModel.from_config(config)
 
     return config, model
+
+
+def custom_sigmoid(output: Dict):
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/models/utils.py:550]: How about to change the name to `apply_sigmoid`?

Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/models/utils.py]: Thanks! Updated.

**Comment by sxjscience** [comment]
Noticed a conflict about ` multimodal/src/autogluon/multimodal/models/utils.py ` . The approval is conditioned on 1) conflict has been resolved, 2) PR has passed the CI.

**Review by bryanyzhu — APPROVED** [review]
Looks good, thank you. Maybe you want to merge the utils refactor PR first.

**Review by suzhoum — APPROVED** [review]
LGTM!
